test/gtest/uct/test_md.cc

```diff
@@ -288,6 +288,61 @@ UCS_TEST_P(test_md, mem_type_detect_mds) {
     }
 }
 
+UCS_TEST_P(test_md, mem_query) {
+    for (size_t i = 0; i < mem_buffer::supported_mem_types().size(); ++i) {
+        ucs_memory_type_t mem_type = mem_buffer::supported_mem_types()[i];
```

Comment:
maybe cache the result of `mem_buffer::supported_mem_types()`?
otherwise, it will do a copy of the `std::vector` every time it is called

Reply:
it's static: https://github.com/openucx/ucx/blob/master/test/gtest/common/mem_buffer.cc#L48

Reply:
but it is copied from static `std::vector` to the `std::vector` that is the return value every time as `supported_mem_types()` called. or am I missing smth?

Reply:
good catch - added return by reference